packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -40,7 +40,7 @@ class RenderProxyBox extends RenderBox with RenderObjectWithChildMixin<RenderBox
   /// Proxy render boxes are rarely created directly because they simply proxy
   /// the render box protocol to [child]. Instead, consider using one of the
   /// subclasses.
-  RenderProxyBox([RenderBox child = null]) {
+  RenderProxyBox([RenderBox child = null]) { //ignore: avoid_init_to_null
```

Comment:
My first guess would be that this is related to strong mode and type inference, though I don't know why the presence or absence of a default value impacts it. Notice the difference in signatures:

'RenderTransform.applyPaintTransform' ('(**RenderBox**, Matrix4) → void') isn't a subtype of 'RenderProxyBoxMixin.applyPaintTransform' ('(**RenderObject**, Matrix4) → void')

@leafpetersen Any insights? Am I off-base with my guess?